### ISSUE TITLE
Slicer creates an additional clearly perceptible and visible Z seam

### ISSUE THREAD

**Issue by Hamsterberg** [SoftFever/OrcaSlicer#7326]
### Is there an existing issue for this problem?

- [X] I have searched the existing issues

### OrcaSlicer Version

2.2.0

### Operating System (OS)

Linux

### OS Version

Linux Mint 21.3 Virgina

### Printer

Bambu Lab P1S

### How to reproduce

slice and print the file with "scarf joint seam (beta) is enabled

### Actual results

slice and print the file with "scarf joint seam (beta) is enabled, an additional clearly perceptible and visible Z-seam is printed

with version 2.1.1 the additional clearly perceptible and visible Z-seam is not present

### Expected results

No clearly perceptible and visible Z-seam at this position

### Project file & Debug log uploads

[test_v2-2-0.zip](https://github.com/user-attachments/files/17596821/test_v2-2-0.zip)

### Checklist of files to include

- [ ] Log file
- [X] Project file

### Anything else?

![back](https://github.com/user-attachments/assets/48ceaebd-db66-458e-ac5b-704ebe95d0eb)

![front](https://github.com/user-attachments/assets/d22e298f-450f-4d6d-94ee-4acc05c8cfa3)

![Slicer preview with z-seam](https://github.com/user-attachments/assets/68cb337c-3e37-4d23-8c0b-04c3171c6e3d)

**Comment by xantrk** [comment]
I think this has something to do with pressure equalizer. When it's off, flow change is abrupt, so it creates an additional "seam".

When it's one, speed is weird, which again creates another seam worse than scarf.

![image](https://github.com/user-attachments/assets/c000a5a9-1b15-4a1a-8537-369bb234c0c6)
^Pressure eq off.

![image](https://github.com/user-attachments/assets/a3681276-1a71-47bb-9f8e-15963ae1064a)
^Pressure eq on.


Scarf speed seems to override the equalized speed.

**Comment by bdwilson** [comment]
Same here. Bambulab A1.  2.1.1 scarf was tuned after much trial and error.  Nothing changed on 2.2.0. All variables look the same in preview.  Flow was the only one that looked a bit off.  

![IMG_9057](https://github.com/user-attachments/assets/ccf38b92-85a1-4dfb-9ffd-036b761354a8)

<img width="1038" alt="image" src="https://github.com/user-attachments/assets/5e08b5d8-4601-4854-b93b-28422c7815b2">

**Comment by zarbam** [comment]
I'm having the same effect on my macbook on 2.2 but fine on 2.1 (windows)

**Comment by shyblower** [comment]
Me too

**Comment by Sidiusz** [comment]
Yep, same here
![image](https://github.com/user-attachments/assets/2ec64623-8fc0-4c9f-bc3a-17d14a79e43c)
(scarf seam on the right)

**Comment by shyblower** [comment]
Today I found out that in 2.2.0 you need to activate "Retract on layerchange", "Wipe while retracting" and a wipe distance of about 0.8mm to make scarf seams look as they did in 2.1.1 where activating "Wipe while retracting" made it look worse.

**Comment by Hamsterberg** [comment]
![Bildschirmfoto zu 2024-11-30 11-46-32](https://github.com/user-attachments/assets/0b6e3c10-a0d8-4200-b392-b77685f40619)

However, your suggested settings are already activated by default in my printer profile.

**Comment by shyblower** [comment]
> However, your suggested settings are already activated by default in my printer profile.

I had it deactivated in the previous releases since it used to work much better without but now it seems to work only when it's activated. I'll post an example when I get to my printer.

**Comment by shyblower** [comment]
This is what it looks like, now that I've reenabled "retract on layer change" and "wipe while retracting" with 1mm wipe distance. On 2.1.1 I had to disable wipe to achieve a comparable result.
https://github.com/user-attachments/assets/337b9d2c-9b89-4d7f-bf22-73533a393ac9

**Comment by xantrk** [comment]
Could it be Z-Hop? https://github.com/SoftFever/OrcaSlicer/issues/7380

**Comment by shyblower** [comment]
Now even "Wipe while retracting" stopped working for me. Don't know why.
Now I also get the this clearly visible seam every time, like all the others here.

**Comment by shyblower** [comment]
Seems to be caused by the priming (unretract) before starting the scarf slope. The length is mathematically calculated perfectly as it is equal to the retraction length but it seems that during retraction a very small amount of already printed material from the end of the line gets also sucked back into the nozzle and thus is pushed out again at the beginning of the next line. This amount is so small that it does not matter too much when the next line is drawn with the same vertical thickness as the previous one (the configured layer height) but since the nozzle starts the next line very closely positioned over the previous layer when the scarf seam is applied,  this unintentionally carried and unretracted extra material produces the blob since it has to expand far more in horizontal direction due to the very small gap between nozzle and the layer below.

This is a hypothesis that I have put forward while testing different configurations and the only one that produced perfect scarf seam was outer/inner wall printing order, no infill and NO retraction during layer change. This way the nozzle moved without retraction from the end of the previous layers inner wall to the next layers outer wall scarf seam start.
Doing anything that needed "unretract" directly before starting the scarf, produced the blob.

**Comment by shyblower** [comment]
I seem to be following the right path because configuring a negative "extra length on restart" value in the printer settings or via the setting overrides in the filament settings thus effectively reextruding less distance then was retracted improves the situation.
But when you set this value too low, air bubbles will be mixed into the beginning of the next line which will result in the same kind of issue you get yourself in if your nozzle is oozing molten filament during travel moves.

I just wonder why it seemed to work perfectly before v2.2.0?

I've tested all this using PCTG filament. It may of course behave differently when using eg. PLA.

**Comment by shyblower** [comment]
What currently works best for me with scarf seam is, inner/outer wall printing order and "retract on layer change" disabled but you have to ensure that if the layer change happens inside the same object that the "travel distance threshold" is configured high enough to not force retraction.

**Comment by begna112** [comment]
@shyblower what kind of values are you using for the negative extra length and travel distance threshold? I got a very interesting/bad result with -1 and 1. 
![20250102_042346](https://github.com/user-attachments/assets/21ff676f-579f-472c-bd3d-fcfdcc3e6110)

**Comment by xantrk** [comment]
Prusaslicer creates good scarfs though with identical setings, so I think this is a bug

**Comment by begna112** [comment]
So I've been able to confirm your results, @shyblower. 

Sunlu PLA+, X1C, .4 high flow nozzle, inner outer inner.

I've found that setting the negative extra length to half the line width size (.42mm/2 = .21mm) gives nearly perfect results on outer walls. However, with scarf seam enabled for inner walls, this leads to noticeable gaps. Notably, my pressure advance is also .2, so it's possible that's the value it corresponds to, not line width.

![20250102_072906](https://github.com/user-attachments/assets/c741d7c0-f19d-48cd-b960-561e30cf0e0f)

(Without retraction and deretraction, I'm getting stringing that I otherwise wouldn't, but that makes sense.) 

Disabling scarf seam gaps seems to mostly negate the inner scarf issue and lead to better quality. Almost imperceptible seam without flash enabled and barely able to feel it. 

![20250102_080720](https://github.com/user-attachments/assets/86df44e0-5b67-452c-b5f6-9625680ed641)
(Left with Seam Gap 10%, right with Seam Gap 0%)
![20250102_074955](https://github.com/user-attachments/assets/c5b0da59-dde7-4761-86ed-04e7155f1568)
(Left with Seam Gap 10%, right with Seam Gap 0%)
![20250102_074958](https://github.com/user-attachments/assets/b8abde06-0855-4907-ac89-425e098eb1b5)
(Left with Seam Gap 10%, right with Seam Gap 0%)
![20250102_080523](https://github.com/user-attachments/assets/7871376d-c133-4f97-87cf-09e4176053e2)
(Left with Seam Gap 10%, right with Seam Gap 0%)

Caveats: 
* inside vase concave shape (overhang on outside of wall) still has a seam, but it's not the straight blob one from before. 
* Overhangs are under extruding a bit. Maybe cause inner/outer/inner. If I remember correctly, that was already a caveat of scarf seams.

**Comment by shyblower** [comment]
@begna112 you only have to deactivate "retract on layer change", the retraction length should of course not be touched.

**Comment by begna112** [comment]
> @begna112 you only have to deactivate "retract on layer change", the retraction length should of course not be touched.

Yes, I know. But for non-scarf seams, retracting on layer change results in best quality and 0 stringing _for my filament and printer_. Just a caveat for my tuning that others might also have.

**Comment by dodox1** [comment]
This is my test print with scarf seams activated.
I tried to modify the G-code (line 3581):
...
G1 X102.71 Y112.381 Z2.8 E.00008
G1 E-1.5 F3600
**;G1 Z3.2 F9600 ; dodo change - disabled**
G1 X103.507 Y116.973 Z3.2
G1 Z2.8
...

So, there is Z movement simultaneously with X and Y movement, and this scar almost completely disappeared.
I'm not saying it is a universal and simple solution, but it just shows what could be the possible reason for that scar.

![PXL_20250106_195049990~3](https://github.com/user-attachments/assets/1c0f1ea9-4f09-4160-a226-a4a11454984f)
![orca_seam_fix](https://github.com/user-attachments/assets/767af142-c9d6-4984-8a23-da24586dd58d)
[tesst_obloucek_PLA.gcode.gz](https://github.com/user-attachments/files/18324157/tesst_obloucek_PLA.gcode.gz)

**Comment by shyblower** [comment]
@AngryMammoth I've already found the git commit that's responsible for the mess. This commit tries to fix the bug where the nozzle moves through already printed stuff when it's traveling to the starting point of a scarf seam, but obviously does it improperly.
I haven't had time yet, though to look deeper into it and fix it.

**Comment by begna112** [comment]
@shyblower could you link it? Would be interesting to see.

**Comment by shyblower** [comment]
@begna112 https://github.com/SoftFever/OrcaSlicer/commit/cf6d9c7

**Comment by shyblower** [comment]
@dodox1  https://github.com/SoftFever/OrcaSlicer/commit/cf6d9c7 is the reason why this happens. This commit tries to fix the problem where the nozzle would move through already printed parts when descending to the scarf seam start position during a travel move but also causes exactly the problem you've described.

**Comment by dodox1** [comment]
What I showed is not that the nozzle travels through the printed material, but instead travels upwards on the spot.

**Comment by shyblower** [comment]
@dodox1 yes and exactly this is accidentally caused by the commit I've linked, but only when a sloped or spiral z-hop is done.

I'm currently trying to fixing it in my own OrcaSlicer fork and will post a pull request here if I'm successful.

**Comment by shyblower** [comment]
@dodox1
 Although the referenced commit actually fulfills its task correctly for normal z-hops, namely to prevent the nozzle from being driven through objects that have already been printed, it also causes the problem you describe due to the intentionally separated z movement.

It doesn't seem to become easy for me, though, to combine x,y and z travel to the scarf seam starting point in a downwards sloped movement and also prevent the nozzle from moving through already printed stuff.

**Comment by shyblower** [comment]
@dodox1 and there is still the problem, I've mentioned somewhere above, where obviously some molten filament from the previous line end gets accidentally sucked in during retraction which is spit out again at the beginning of a new line. This is not so obvious when a normal line is printed, because of the relatively large cross section compared to the start of a scarf seem line.

**Comment by nubnubbud** [comment]
I have also been experiencing this problem. I don't think it's due to retraction, totally if at all. I was able to reduce it by increasing jerk settings, then increasing seam gap. This is obviously the wrong solution, but it seems more likely to be something like oozing mid-retraction, and the extra volume escaping during de-retraction. retraction, in my experience, doesn't "suck" as much as it simply relieves pressure.

I am confident this bug is why "outer first" is suddenly a preferred way to print like this, and what I have been forced to use, as anything else triggers the "fix" to reduce 

I have an old CR-10S, and I can confirm it's happening, even with a bowden tube, and I even made a new firmware to test pressure advance, to no avail- it doesn't seem to be that, either. Interestingly, if it was just a re/detraction issue, I imagine a bowden tube setup would be getting different results, but I'm seeing a very disctinct, uniform bulge at the beginning of the scarf, just like a P1S. 

until this is solved, is perhaps the best option to rollback to 2.1.1? the 2.1.1 version, however, has a number of incompatibilities with other features... **for now, scarf seam users might be best off using the 2.2 RC version.** the full release had the change in its changelog, and the biggest change otherwise I can tell is the new PA calibration pattern.

as for a correct way to move the nozzle down, it should begin tracing along the perimeter it's about to start printing, and then dip down along it to the intended scarf joint start without slowing. This would solve several issues;
1. speed/flow variation issues as the nozzle goes from still at the beginning of the scarf joint to its end
2. sudden angle changes as it moves from a different perimeter to the outer one
3. by tracing the end of the perimeter it is about to print, you can guarantee the nozzle is only moving along an area without filament laid down.

but I might be missing something here.

**Comment by nubnubbud** [comment]
I just did a small battery of tests. Even with an entire millimeter of extra advance after a retract, it did nothing much. https://github.com/SoftFever/OrcaSlicer/commit/cf6d9c7 is clearly the issue, as version 2.2.0-RC produces perfect results on my 7 year old CR-10S with custom firmware. Pressure advance had no major effect on it, nor did any retraction settings or wipes, but going to version 2.2.0 release version causes the issue again. 

If you use scarf seams, use the release candidate for now as a workaround. I don't find any changes between the release candidate and full release compelling enough to justify broken scarf joint seams for most users.

**Comment by xantrk** [comment]
@SoftFever

**Comment by shyblower** [comment]
@nubnubbud I'm also quite sure it is this fix that introduced the difference in behavior between 2.2.0 and 2.1.1.
I also have an explanation for this:

When the nozzle leaves the end of a line for a travel move to the next line then a tiny bit of molten filament stays on the nozzle tip, even if it did a wipe movement before lifting off (I've observed that when the nozzle leaves an object for another wipe moves are often carried out in the opposite direction than the line was drawn, effectively gliding over still molten material). Retraction also might have sucked in a fraction of the still molten plastic underneath it, as I've already suggested somewhere above.

When z-hop is done before starting the scarf slope e.g. when you print outer/inner wall order or "retract on layer change" is off and z hop height is greater then zero and also with with the fix in https://github.com/SoftFever/OrcaSlicer/commit/cf6d9c7 in any case, the nozzle is moved to the scarf slope start position in a safe distance above the previous layer and then lowered down to the start z position which is with the default of 10 scarf steps one tenth of the layer height. This flattens out the tiny drop of filament and expands it horizontally creating a blob which consequentially leads to the scar like seam.

Without the above mentioned fix when your wall order is inner/outer then the nozzle gets dragged through the already printed layer because the the scarf slope starts, as described above, at a lower position than the layer height. During this the molten remnants from the previous line get completely wiped of the nozzle and no blob is produced.

Also the above mentioned fix for the nozzle being dragged through the layer is not properly implemented since it only works correctly for legacy z hops but not for sloped or spiraled ones.

**Comment by nubnubbud** [comment]
@shyblower I disagree. I was printing tiny dense rollers, and watching. there is no room for anything under the nozzle. I don't use z-hop or anything like that, and wiping didn't affect it either. it seemed that the nozzle just stayed there, extruding for a moment, before starting the scarf seam.

I never mentioned z-hops for a reason, they were never a consideration in my testing, and my suggestion did not include or consider them. if your printer is properly calibrated and your profiles are well made, z-hops should do pretty much nothing, as they exist to fight oozing and stringing from machines that don't have accurate extrusion or can't accommodate material oozing.

My solution was only to make sure, that no matter what, the scarf seam began in the middle of a continuous movement along a perimeter that was guaranteed not to have any extrusion, therefore eliminating the need for advanced algorithms, z-hop mechanics, and changes to the travel code altogether. Keep It Simple Stupid.

**Comment by shyblower** [comment]
@nubnubbud then it's the additional molten filament, that gets sucked in by retraction from the still liquid end of the line and is reproduced during unretraction. That's why I could reduce the issue with a tine negative extra prime length.

**Comment by nubnubbud** [comment]
@shyblower ah, perhaps. I still find that suspect, but I use a bowden, so for me retraction is all about eliminating pressure in the bowden tube and eliminating oozing right at a travel. Maybe that's why you would use z-hop and I can't- for me, because mine can only reduce the residual bend inside the bowden tube, not z-hopping is how I eliminate stringing, but a faster, less spring-like reaction might allow yours to actually suck. (...in the good way.)
also of note, I turned my retraction and pressure advance settings against the seam, and there was zero space where I would get somethng reasonable. it went straight from positive seam, to at least half a millimeter of missing wall. I do pride myself on my calibration, so my machine shouldn't be sloppy enough to jump around the .3mm in extrusion that can happen in.

regardless of the mechanism (though I suspect it's something along the lines of "it begins extruding for the scarf joint too early because of pressure advance taking effect before the scarf joint sets extrusion to almost nothing" or "it should be de-retracting as the move starts, not before") we can agree that it's extruding when it shouldn't, and it's because of that change, but we're not sure why.

**Comment by shyblower** [comment]
@nubnubbud and the reason why it worked so well for many people in 2.1.1 because of the issue of the nozzle being dragged through the previous layer in certain configurations which effectively cleaned the nozzle and might even have pulled out the remnants of the slightly sucked in material of.the previous line end.

That's why it works again when I remove the above mentioned fix for this which is implemented in 2.2.0, and is not working correctly anyway when used with sloped or spiralled z hop.

**Comment by xantrk** [comment]
@shyblower Do you know if prusaslicer implementation is similar? That works perfecty

**Comment by shyblower** [comment]
@xantrk if PrusSlicer uses the same Implementation it's probably still working because they haven't tried to fix or even still not recognized the nozzle getting dragged through the previous layer issue.

**Comment by SoftFever** [comment]
@shyblower Nice work on the tracing and investigation!  Have you made any progress on the fix by any chance? 

Meanwhile, I pull the PR's author @fritzw in the discussion

**Comment by fritzw** [comment]
Interesting, and good detective work. I also noticed this but I didn't think to go back and check if it could have something to do with my fix.

Just reverting the fix seems to be a good solution for small models, but for larger models with multiple separate perimeters (think e.g. a filament spool with cutouts in the side wall), the original behavior causes major collisions and even print failures, as the nozzle drags through the previously printed perimeters with quite some force. At least on my printer.

Maybe a proper solution would be to do the downward move slightly inside the model (on the second or third outermost perimeter) next to the spot where the scarf seam should start. This would wipe the tiny blob on the second perimeter instead of the outer one, hopefully reducing the effect. Maybe additionally offsetting the move-down point 2-3 mm "backwards" (i.e. tracing the perimeter it is about to print like @nubnubbud suggested) could also help by reducing the sharp corner at the start of the scarf compared to just moving the move-down point inwards orthogonally. I hope that explanation makes sense...

**Comment by nubnubbud** [comment]
@fritzw Well, I had several ideas. I'm not sure about the exact toolpath the code makes, but I imagine it could be turned from one angled line, to two lines- a normal travel, and a diagonal path for the last few millimeters. 

but I keep reiterating- I really am unsure of where the extra material is coming from (and why I keep grilling Shyblower on being sure something is the case without experiments), and unlike the others, I'm not willing to guess exactly why more filament comes out by the time the nozzle has begun the ramp. it could be;

- Jerk settings cause it to slowly come to speed, while extrusion is immediate, meaning too much initial extrusion.
- the nozzle is oozing before it lands (despite retractions or other things. are they taking effect or not?).
- it's probably not linear advance- I tested it and there were no points where it fixed anything.
- oozing could happen internally, and unretraction could push it out.

and probably some others.

**Comment by igiannakas** [comment]
@fritzw @shyblower would the wipe inside before external perimeter help here? It lowers z and unretracts 1 nozzle width inside the model, then moves to the outer perimeter and starts printing. 

This wipes the nozzle inside the model before moving externally.

**Comment by igiannakas** [comment]
This option here: https://github.com/SoftFever/OrcaSlicer/pull/3287

**Comment by fritzw** [comment]
> This option here: https://github.com/SoftFever/OrcaSlicer/pull/3287

Nice, that sounds like what I tried to describe. Do you know how this implementation handles features that consist only of two outer perimeters (i.e. one perimeter loop)?

**Comment by igiannakas** [comment]
> > This option here: [#3287](https://github.com/SoftFever/OrcaSlicer/pull/3287)
> 
> [@igiannakas](https://github.com/igiannakas) Nice, that sounds like what I tried to describe. Do you know how this implementation handles features that consist only of two outer perimeters (i.e. one perimeter loop)?

I think if I recall correctly, it will not execute for a single wall perimeter - take a look here: https://github.com/SoftFever/OrcaSlicer/pull/3616

**Comment by shyblower** [comment]
@igiannakas, @fritzw  Yes, sounds definitely like what both of us were suggesting. I'll have to try it.

**Comment by shyblower** [comment]
@igiannakas, @fritzw, @SoftFever Unfortunately "Wipe before external loop" does not help, in fact it looks the same as if done with "correct" z-hjopping and I believe I know why. Take a look at the attached picture and you will see.

I tried this with my build where I've removed the z-hop fix so that the nozzle moves through the layer as in 2.1.1.
Both with retract on layer change and thus z-hopping in this case deactivated.

Upper: no wipe before external loop
Lower: wipe before external loop

Imho it works in the upper example because the nozzle moves directly in a slope down to the start of the scarf thus effectively wiping the nozzle and starting the scarf without having to move down the nozzle in a vertical move directly over the starting point, thus not squashing any filament remnants into a blob.

In the lower example, with "Wipe before external loop" activated, it does the backwards wipe move, collecting material from the still liquid end of the line, carrying it in a straight horizontal line above the scarf start, lowering the nozzle vertically and thus squashing the tiny bit of collected molten filament into a rather large blob, because of the very low height of the first scarf segment.

![Image](https://github.com/user-attachments/assets/ee8762cd-3eab-4459-b9be-1279edd4f5f3)

**Comment by igiannakas** [comment]
What’s puzzling me is how large that blob is. What does the seam look like without scarf?

**Comment by nubnubbud** [comment]
at this point I would be asking for a slow motion macro shot of it as it happens. it could be that the nozzle presses too far down and ends up remelting and bulging the wall, or any number of things. we need better observations.

**Comment by Sidiusz** [comment]
bump, the problem still exist

**Comment by GandalftheGay** [comment]
any fix/good workaround for this yet?
just did many tests because i could not find good settings without the the left start of the scarf seam bulging. Upping the "Scarf start height" made it even worse while making the middle of the seam nearly invisible (=

if i can somehow help to fix this problem with testing, just tell me

**Comment by shyblower** [comment]
The only thing that works for me was reverting the fix (that had been implemented since 2.2.0) for preventing the nozzle to move through already printed parts of the current layer when moving to the start of the scarf slope and deactivating "retract on layer change" to force exactly this behaviour which effectively wipes the nozzle clean and thus prevents the blob at the start of the scarf which is the definite cause of that seam.

Since the nozzle moving slightly through the current layer has never caused any harm to my prints yet as long as I activate retraction for the top layer, I'll leave it that way for now until I or someone else comes up with a better solution 🤷🏼‍♂️

**Comment by zarbam** [comment]
> any fix/good workaround for this yet? just did many tests because i could not find good settings without the the left start of the scarf seam bulging. Upping the "Scarf start height" made it even worse while making the middle of the seam nearly invisible (= i used PLA and ASA for all my tests btw.
> 
> if i can somehow help to fix this problem with testing, just tell me
> 
> also: would be nice to have an option to activate wipe before external loop for normal seams while having Conditional scarf joint enabled.

Not yet that I know of. I'm stuck on 2.1 because of the products I sell rely on this to work

**Comment by bdwilson** [comment]
> Not yet that I know of. I'm stuck on 2.1 because of the products I sell rely on this to work

Same here. Stuck in time and missing the new features.

**Comment by daithi-coyle** [comment]
Nothing to add really other than another data point.
Getting my best results (still with a seam but at least not a second seam/bulge) with "Scarf: OFF", "Walls Printing Order: Inner/Outer/Inner"

![Image](https://github.com/user-attachments/assets/aaf83826-24e1-46d5-9698-979648702582)

I have not tried Scarf: On with "Wipe Before External", "Wipe on Loops" and reducing "Wipe Speed" but based on the above comments that won't make much difference.

**Comment by pm0u** [comment]
Is there a tldr; on this? i did read but its a bit hard for me to follow. Is v2.1.1 the answer? Was the offending code reverted for 2.2.0/2.3.0?

**Comment by daithi-coyle** [comment]
> Is there a tldr; on this?
Still an issue 

> Is v2.1.1 the answer?
It's the easiest solution 

> Was the offending code reverted for 2.2.0/2.3.0?
Not so far

**Comment by pm0u** [comment]
@daithi-coyle much appreciated, I'll give 2.1.1 a shot and see what I get.

**Comment by igiannakas** [comment]
@SoftFever @Noisyfox what do you think here? Should the z change be reverted? Or disabled when scarf seams are on?

**Comment by shyblower** [comment]
@igiannakas I just reverted the z-hop fix (that actually does not even work correctly with spiralling and sloped z-hop) in my 2.3.0 branch some weeks ago and haven't been experiencing any issue so far.

**Comment by pm0u** [comment]
I'm actually a bit confused - i went to do the same as @shyblower but the commit was unable to revert when i attempted. It looks like it already was, ahead of 2.3.0?

https://github.com/SoftFever/OrcaSlicer/pull/8478

Is there some other added code that is also contributing to this issue?

**Comment by pm0u** [comment]
Results of my testing of 2.3.0 vs 2.1.1 - I don't see any difference myself. I can share the generated GCode if desired. This is the same project, created in 2.3.0 and then imported into 2.1.1. It's possible there's a couple settings that weren't usable in 2.1.1 hat could contribute a tiny difference. I did disable adaptive pressure advance for the 2.3.0 print to rule that out.

pics are incoming...

**Comment by shyblower** [comment]
@pm0u you have to disable "retract on layer change" for the best result on 2.1.1 or own builds after reverting the z-hop fix which has introduced the issue.

**Comment by ssmb212** [comment]
> [@pm0u](https://github.com/pm0u) you have to disable "retract on layer change" for the best result on 2.1.1 or own builds after reverting the z-hop fix which has introduced the issue.

A lot of reading, we should have a pinned answer maybe with current up to date latest stable release manual fix for this.

To my understanding if you want seams to work on 2.3 main channel you need to do the following.

Retract on layer change - Disabled
Z hop - normal
Wall order inner outer or inner outer inner (2-walls minimum respectively)

Unsure for the following

Wipe while retracting - disabled/enabled ?
extra length on restart -0.05 necessary?
wipe distance 1mm untouched?
retract amount before wipe any change here?
Z hop completely disabled?

Would different z hop heights work differently ?
Would different retraction detraction speed work differently ?


Thanks in advance for the answer and the already effort put.

**Comment by ssmb212** [comment]
> > [@pm0u](https://github.com/pm0u) you have to disable "retract on layer change" for the best result on 2.1.1 or own builds after reverting the z-hop fix which has introduced the issue.
> 
> A lot of reading, we should have a pinned answer maybe with current up to date latest stable release manual fix for this.
> 
> To my understanding if you want seams to work on 2.3 main channel you need to do the following.
> 
> Retract on layer change - Disabled Z hop - normal Wall order inner outer or inner outer inner (2-3 walls minimum respectively)
> 
> Unsure for the following
> 
> Wipe while retracting - disabled/enabled ? extra length on restart -0.05 necessary? wipe distance 1mm untouched? retract amount before wipe any change here? Z hop completely disabled?
> 
> Would different z hop heights work differently ? Would different retraction detraction speed work differently ?
> 
> Thanks in advance for the answer and the already effort put.

Anyone care to enlighten ?

**Comment by nubnubbud** [comment]
> > > [@pm0u](https://github.com/pm0u) you have to disable "retract on layer change" for the best result on 2.1.1 or own builds after reverting the z-hop fix which has introduced the issue.
...
> 
> Anyone care to enlighten ?

I'm unsure too. I thought the change that caused the issue was reverted before 2.3 so I have no clue where we stand, what works, or what doesn't...
...but given the hard-to-follow thread here, it sounds like 2.1.1 is still the only version that works without any exceptions.

> > Is there a tldr; on this?
> 
> Still an issue
> 
> > Is v2.1.1 the answer?
> 
> It's the easiest solution
> 
> > Was the offending code reverted for 2.2.0/2.3.0?
> 
> Not so far

this is the closest to a definitive answer we currently have. "just go back to 2.1.1, it's not fixed."

It makes no mention of the other features this borked for use with scarf seams, leaves them unknown and undocumented in the software, and so on. If you'd like to test which features have been negatively affected, please do! we need this info. just... please use a neutral color filament like gray, sky blue, or so on. White, high intensity colors, and black are terrible for visualizing issues, as seen above.

it seems like the code hasn't ever been reverted. From what I can gather, they're trying to fix the many issues and complexities it added, rather than implement something different that will do the same thing, but there's been no real paper trail here as to which commits or so on have been added or reverted. I've made some suggestions, did a week of testing and so on, but there's been no response to most of my efforts or new data, so I've sorta just decided to give up on expecting scarf seams to work after 2.1.1. 

two main releases later, the offending change has not been reverted, so for now it's probably just a permanent part of the software via technical debt now I guess. maybe someone will get angry and revert it after everyone else forgets why it was changed in the first place.

**Comment by shyblower** [comment]
For the best results you must not use outer/inner wall printing order and also disabling "retract on layer change" lay help because then no z-hop is done during layer change and the nozzle will wipe itself clean by scraping over the inner wall perimeters before starting to print the scarf slope.
In any case, you must prevent the nozzle from starting the outer perimeter scarf arriving directly from a travel move, without the possibility to wipe off that tiny droplet from its tip that seems to be always present after having z-hopped at the end of the previous line, regardless of any wipe movement or z-hopping method. The nozzle wiping at the end of a line might even add more material to the nozzle tip.

The reason why this tiny drop of molten filament causes such a large horizontal blob at the start of a scarf seam is, because the layer height at the beginning of the scarf is very low. 0.02mm if you print with 0.2mm layer height and 10 scarf steps.

**Comment by ssmb212** [comment]
Hi thanks both for your answers. I have been doing extensive testing both with pla and tpu (different effects and desirable results on both of course).  Last time i used orca 2.1.1 or so was working fine and making nicely looking seams. I went through an apartment move and for a long period my printers where not set up. After setting up and downloading latest orca slicer went through and just recreated profiles (i got new printers sold some old etc, saw no point of transferring old profiles). 

From this point on practically i get the double seam effect. googling it led to a reddit thread which led here. And the pictures of the double seams are identical to the result i get (day and night difference of what i was printing before). After reading the whole conversation here i startd everything through scratch made sure to re calibrate flow ratio temperatures pressure advance etc just to be sure. And afterwards i started testing changing a setting at a time trying different combinations etc. I cant say it hasn't improved but its still there practically its between having bulbous seam and a double ghosting like seam further as seen on the pictures or having a mid way regular usual seam like scarfs are off.

If it wouldn't be too much of a trouble can each of you share an vanilla orca 2.3 .3mf with your settings ? Doesn't really matter which printer got most of the main band ones so i can see if ill get identical results figure out what the issue in my case etc. I will post results afterwards as best as i can, sadly i didnt kept the samples organized just skipping through yes/no changes and pilling up in a corner but i can do a better work until i get the desired results.

**Comment by pm0u** [comment]
I won't be at my computer for a few days, but I can pull up the profile from those test prints or another one from recent.

This was a print in ABS, 2.3.0, scarf enabled. printed fast - max 24mm³/s, 200mm/s outer walls, 40% scarf speed, z hop on retraction 0.4mm, retract on layer change, 0.2mm layers. I'm not sure if I should hope for better but I thought this was a pretty good result.

![Image](https://github.com/user-attachments/assets/8267faea-dc42-402c-a945-bfad7b4c421e)
![Image](https://github.com/user-attachments/assets/8bdafbae-0517-4e65-b036-d311cecf0330)
![Image](https://github.com/user-attachments/assets/f8699347-18d6-45e2-9c50-31263b5f42f9)

**Comment by ssmb212** [comment]
> I won't be at my computer for a few days, but I can pull up the profile from those test prints or another one from recent.
> 
> This was a print in ABS, 2.3.0, scarf enabled. printed fast - max 24mm³/s, 200mm/s outer walls, 40% scarf speed, z hop on retraction 0.4mm, retract on layer change, inner/outer walls 4 walls, 0.2mm layers. I'm not sure if I should hope for better but I thought this was a pretty good result.
> 
> ![Image](https://github.com/user-attachments/assets/8267faea-dc42-402c-a945-bfad7b4c421e) ![Image](https://github.com/user-attachments/assets/8bdafbae-0517-4e65-b036-d311cecf0330) ![Image](https://github.com/user-attachments/assets/f8699347-18d6-45e2-9c50-31263b5f42f9)

This looks great, pretty much it looks how i was printing back in the summer. Dont know what changed since then i need to try to figure it out. Most of my settings fit what you said although i print on 3 walls, wonder if this could make so much difference?? anyhow i will wait for the profile.

**Comment by bdwilson** [comment]
For anyone who needs to hear it, I've not had luck on anything beyond 2.1.1.  I grabbed testsuite_orcasliceronly_v6.3mf from here: https://makerworld.com/en/models/211686-better-seams-orca-slicer-guide-to-scarf-seams

I did a few tests and they worked flawlessly on my previously flawed prints.  ChatGPT compared the before and after. Here are the biggies that needed changing. 

This works great on both my Snapmaker U1 (Using Snorca) and my Bambu A1 printers using latest Orca. 

1) Conditional scarf seams were enabled (this is the biggest one)

Old profile (2.1.1 style behavior)
	•	seam_slope_conditional = 0

New profile (working 2.3)
	•	seam_slope_conditional = 1

This means:
	•	old → scarf always tried to ramp regardless of geometry
	•	new → scarf only activates when the perimeter is long enough & stable

This prevented the scarf from trying to taper right in the tight waist area, which is exactly where PETG-CF and Snapmaker were tearing fibers / producing razor seams.

⸻

2) Scarf now applies to inner walls too (stabilizes pressure)

Old
	•	seam_slope_inner_walls = 0

New
	•	seam_slope_inner_walls = 1

Why this helped:
	•	before → only the outer wall was scarfed
	•	inner walls had hard stop/restart
	•	pressure oscillated, outer wall suffered

Now:
	•	all walls follow the same taper event
	•	pressure drops smoothly
	•	outer wall scarf stops looking fat/chewed at the beginning

This one change reduces the “fat at scarf start, fine at end” symptom significantly.

⸻

3) You doubled the number of scarf steps (more gradual taper)

Old
	•	seam_slope_steps = 10

New
	•	seam_slope_steps = 20

Meaning:
	•	more but smaller reductions
	•	melt pressure doesn’t “fall off a cliff”
	•	PETG-CF doesn’t fracture at the first step

This is why the start of the scarf stopped bulging, and texture stopped vibrating.

⸻

4) Scarf joint speed changed (effectively faster)

Old
	•	scarf_joint_speed = 100%  (percent of wall speed)

New
	•	scarf_joint_speed = 100 (fixed mm/s)

On your profile:
	•	outer wall speed is high
	•	scarf joint now runs fast enough to keep pressure up

This is why the new prints match the A1/Bambu look.
Slow scarf = depressurized melt = sharp seam
Fast scarf = preserved pressure = rounded seam

⸻

5) Extrusion-rate smoothing was enabled

Old
	•	max_volumetric_extrusion_rate_slope = 0 (off)

New
	•	max_volumetric_extrusion_rate_slope = 300

This does exactly what your observation implied:

it prevents big, sudden flow changes during scarf ramps

The result:
	•	no more pressure snap-down
	•	no more “crystalline” PETG-CF seam texture
	•	seam loses the knife-edge feel
